libres/CMakeLists.txt

```diff
@@ -82,127 +88,62 @@ conan_cmake_run(
-  # cmake we set it here to location of the to-be-installed site-config file. That
-  # implies that when testing the binary will have embedded a path to non-existent
-  # or stale version of the site configuration file. Irrespective of the value of
-  # SITE_CONFIG_FILE the environment variable ERT_SITE_CONFIG will be set to point
-  # to the in-source version of the site-config file for testing.
-  #
-  # The variable SITE_CONFIG_FILE can be set by the user, if that is set it will
-  # take presedence. If the user has not explicitly set the SITE_CONFIG_FILE variable
-  # we will use the default path based on the install prefix.
-  set( SITE_CONFIG_FILE "${CMAKE_INSTALL_PREFIX}/${SHARE_DIR}/site-config" CACHE FILEPATH "Path to global ERT Configuration file")
-  message(STATUS "The path ${SITE_CONFIG_FILE} will be compiled into the libres library as the location of the site configuration file")
-
-  if (USE_RPATH)
-    SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
-    set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-  endif ()
-endif()
-
-#-----------------------------------------------------------------
-
-set(ERT_LSF_LIB_PATH     "" CACHE FILEPATH "Path to search for the LSF libraries")
-set(ERT_LSF_INCLUDE_PATH "" CACHE FILEPATH "Path to search for the LSF header files")
-find_path(LSF_HEADER_PATH lsf/lsf.h PATHS ${ERT_LSF_INCLUDE_PATH})
-find_library(LSF_LIBRARY NAMES lsf PATHS ${ERT_LSF_LIB_PATH})
-
-#-----------------------------------------------------------------
-
-set(EQUINOR_TESTDATA_ROOT "" CACHE PATH  "Root to Equinor internal testdata")
-if (EXISTS ${EQUINOR_TESTDATA_ROOT})
-    set( LINK "${CMAKE_CURRENT_SOURCE_DIR}/../test-data/Equinor" )
-    if (EXISTS ${LINK})
-        execute_process(COMMAND ${CMAKE_COMMAND} -E remove "${LINK}")
-    endif()
-
-    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink "${EQUINOR_TESTDATA_ROOT}" "${LINK}")
-    message(STATUS "Linking testdata: ${LINK} -> ${EQUINOR_TESTDATA_ROOT}")
-endif()
-
-file(COPY ../job_runner/job_dispatch.py
-  DESTINATION ${PROJECT_BINARY_DIR}/bin
-  FILE_PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ)
-set( ERT_ROOT "${PROJECT_BINARY_DIR}" )
-
-#-----------------------------------------------------------------
-
+find_package(cJSON REQUIRED)
 find_package(Filesystem REQUIRED)
 
 find_package(Threads)
-if (CMAKE_USE_PTHREADS_INIT)
-    set(HAVE_PTHREAD TRUE)
-endif ()
+if(CMAKE_USE_PTHREADS_INIT)
+  set(HAVE_PTHREAD TRUE)
+endif()
 
 # feature tests
 include(CheckFunctionExists)
-check_function_exists( regexec ERT_HAVE_REGEXP )
+check_function_exists(regexec ERT_HAVE_REGEXP)
 
-#-----------------------------------------------------------------
+# -----------------------------------------------------------------
+# Get version from git
+# -----------------------------------------------------------------
+
+set(RES_VERSION
+    "0.0.0-unset"
+    CACHE STRING "libres version")
```

Comment:
I think it would be helpful if the moving if tests, formatting of cmakefiles(with intro if cmake-format), and changes to cmake files (those unrelated to the previous two points), were 3 separate commits.